### DISCUSSION TITLE
Pyboard Availability Update?

### DISCUSSION THREAD

**Discussion by thestumbler** [micropython/micropython#11209] [Hardware & Peripherals]
A couple of months ago, I fried two PyBoards v1.1 because of a cabling error. This left me with zero spares. As I suspected, PyBoards are still not available to purchase. 

Is there any current forecast for future availability? 

I'm 99% sure I could repair the two busted ones by just changing out the MCU, but I suspect that is one of, if not the, component that's holding up the production. 

Otherwise, are there any distributors that sell returns, of course in as-is condition? I could probably cobble a couple together if I could get ahold of a few "broken" ones. 

Worst case I'm going to have to design in another solution. There's a flavor of an RP2040 module that could probably be substituted, but that'd be a big headache. Plus, I really like having a PyBoard (in its metal case) kicking around in my toolbox for general purpose debugging.

**Comment by peterhinch** [comment]
The problem is zero availability of the STM chips. I don't know what the current lead time is.

**Reply by thestumbler** [reply]
Yeah, as I suspected.

**Reply by thestumbler** [reply]
Wow, I hit pay dirt! I was meeting with an old client yesterday about a small board they want me to design. Their latest products are using the STM32H7 MCU, another difficult to buy chip. I asked what MCU they preferred for this small PCB, since it seemed that an H7 was way overkill. To my surprise, he left the meeting and came back with a tray full of STM32F405 chips we had used on a previous job years ago. 

Even though we eventually settled on an RP2040 for this new board, he gave me a handful of 405 chips for me to try to repair my PyBoards. 

Surely I woke up on the right side of the bed that day.

**Comment by ThomasAkam** [comment]
@dpgeorge I was wondering if there is any further information available about when pyboard v1.1 will be back in stock.   I noticed that the 
    STM32F405RGT6 microcontroller is now in stock at Farnell (at the link below, 4771 units in stock at £12.2 each for 100+).

https://uk.farnell.com/stmicroelectronics/stm32f405rgt6/mcu-32bit-cortex-m4-168mhz-lqfp/dp/2064363

**Comment by oreoneuro** [comment]
Hi! I came across your post because I'm currently dealing with a very similar failure mode. Two of my PyBoards suddenly stopped working, and the STM32 gets very hot and is no longer recognized over USB.

You mentioned that yours were damaged due to a cabling error. Would you mind sharing what the wiring mistake was? I'm trying to determine whether something similar could have happened in my setup.

Also, were you eventually able to repair the boards by replacing just the STM32F405 MCU, or were other components damaged as well?

Thanks in advance!

**Comment by thestumbler** [comment]
> You mentioned that yours were damaged due to a cabling error. Would you mind sharing what the wiring mistake was? 

@oreoneuro wow, that was a long time ago. I think the problem was I had swapped +5v and ground on a setup powered directly from those pins and not over USB. I think. 

Funny you ask. I still have one of those bad boards in a box and a tray of a dozen STM32s and I never got around to fixing them. But I'm pretty sure I did replace the MCU on one of the bad boards and got it working again. I just can't remember whether any other components needed replacing. If so, it would have been minor ones like diode and/or a resistor. 

My problem was the cable was a four pin connector that, while keyed with a plastic latch one one side, it was possible to mis-plug the cable. And this installation was in a dark, had to reach area of a bigger system and I was plugging it in almost by feel. It was really unfortunate that this happened smack dab in the middle of an STM32 supply draught. I only got replacements a year or two later. 

I use a lot of RP2040 / 2350 these days. But I really miss the nice packaging of the PyBoard with its expansion options and accessible I/O pins.

**Reply by oreoneuro** [reply]
Thanks so much for taking the time to dig up the details from so long ago, @thestumbler — really appreciate it!
Interesting that yours was a reversed +5V/ground on a directly-powered setup. In our case the wiring wasn't reversed, so our leading suspicion is hot-plugging the USB while the board was still powered through the breakout's power jack (possibly compounded by grounding issues). Coming from basically zero electronics background, I had no idea that was even something you could do wrong — lesson learned.
Good to hear you got one working just by replacing the MCU — that's encouraging, because that's exactly the route we're taking now. New pyboards are out of stock everywhere, so we're having our two dead boards repaired by swapping the STM32F405RG (chips from Mouser) through a local rework service. Fingers crossed it revives them.
One more thing, since you mentioned using RP2040 / RP2350 a lot these days — I'm curious how you integrated them with a pyControl setup. Did you have to build custom wiring or adapter boards to connect to the existing breakout, or did you move away from the pyControl breakout entirely? And if you happen to have any design files or notes you'd be willing to share, that'd be incredibly useful — we're weighing alternatives given the sourcing headaches. No worries at all if it's not easy to dig up!
Thanks again

**Reply by peterhinch** [reply]
Chinese clone Pyboards are available on eBay but who knows about their quality :-(
